mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactAudioView.js

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { getSrc } from './ShowArtifactPage';
+import { getArtifactContent } from '../../../common/utils/ArtifactUtils';
+import WaveSurfer from 'wavesurfer.js';
+
+function formatTimecode(seconds) {
+  return new Date(1000 * seconds).toISOString().substring(11, 19);
+}
+
+const ShowArtifactAudioView = ({ runUuid, path, getArtifact }) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState();
+  const [waveformPainting, setWaveformPainting] = useState();
+  const [playing, setPlaying] = useState(false);
+  const [audioDuration, setAudioDuration] = useState(0);
+  const [timeElapsed, setTimeElapsed] = useState(0);
+  const waveformRef = useRef();
+  const [waveform, setWaveform] = useState();
+
+  useEffect(() => {
+    resetState();
+    fetchArtifacts({ path, runUuid, getArtifact });
+    return () => {
+      if (waveform) {
+        waveform.destroy();
```

Comment:
`waveform` is not set the first time this component is rendered, so the very first waveform that gets assigned will never get destroy causing a memory leak. Any tips?